webapp/app/controllers/test.js

```diff
@@ -0,0 +1,69 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  treeCompliantMetadata: function() {
+    var model = this.get('model');
+
+    var metadata = model.get('testMetadata');
+    var compliantObject = [];
+    this.createSubtree(compliantObject, '', metadata);
+    return compliantObject;
+  }.property('model.testMetadata'),
+
+  createSubtree: function myself(array_to_append, parentIndex, treeObject) {
```

Comment:
I'm not sure about refactoring, you are right that it needs to be properly documented (and in addition explaining the treegrid guidelines and what exactly are we doing here)

Indeed it is difficult to understand without the "needed" outcome knowladge

Will give it a shot and then if you still think refactoring is needed - we will have a common ground to discuss about